src/pages/Auth/index.tsx

```diff
@@ -1,40 +1,66 @@
 import { ReactComponent as GoogleIcon } from 'assets/svg/google.svg';
 import { ReactComponent as NaverIcon } from 'assets/svg/naver.svg';
 import { ReactComponent as KakaoIcon } from 'assets/svg/kakao.svg';
-import { useForm, SubmitHandler } from 'react-hook-form';
-import { Link } from 'react-router-dom';
+import { useForm } from 'react-hook-form';
+import { Link, useNavigate } from 'react-router-dom';
 import AuthTitle from 'components/Auth/AuthTitle';
 import Copyright from 'components/Auth/Copyright';
 import cn from 'utils/ts/classNames';
+import { login } from 'api/user';
+import sha256 from 'sha256';
+import makeToast from 'utils/ts/makeToast';
 import styles from './LoginPage.module.scss';
 
-interface IFormInput {
+interface LoginFormInput {
   id: string;
   pw: string;
   checkbox: boolean;
 }
 
+const useLoginRequest = () => {
+  const navigate = useNavigate();
+  const submitLogin = async ({ id, pw, checkbox }: LoginFormInput) => {
+    try {
+      const { data } = await login({
+        account: id,
+        password: sha256(pw),
+      });
+
+      if (data) {
+        sessionStorage.setItem('accessToken', data.accessToken);
+        if (checkbox) localStorage.setItem('refreshToken', data.refreshToken);
+        navigate('/');
+      }
```

Comment:
사소하지만 early return 으로 분기 조건 depth 가 피하는 것이 가독성이 높아지는것 같아요 😅 


```suggestion
      if (!data) {
        return;
      }

      if (checkbox) {
        localStorage.setItem("refreshToken", data.refreshToken);
      }

      sessionStorage.setItem("accessToken", data.accessToken);

      navigate("/");
```

Reply:
그나저나 checkbox 이 boolean 타입인게 어색하네요... 
어떤 문맥에서 체크가 되는것인지는 모르겠지만 `isAutoLoginChecked` 같은 표현이 더 적합하지 않을까요?

Reply:
[요게](https://github.com/BCSDLab/JJBAKSA_FRONT_END/pull/25#discussion_r989210449) 반영되면 !data 조건에 대한 로직은 제거해주면 좋겠네요

Reply:
뎁스가 일정하게 유지되니 가독성 측면에서 좋아보이네요.
checkbox는 제가 해당 부분을 작성하지 않았던지라 신경쓰지 않았는데, autoLoginFlag정도로 이름을 수정하면 좋겠네요.

Reply:
`flag` 란 보단 `is~~` 가 일반적으로 쓰이는것 같아요